packages/firebase_dynamic_links/firebase_dynamic_links_platform_interface/lib/src/method_channel/method_channel_firebase_dynamic_links.dart

```diff
@@ -0,0 +1,192 @@
+        case 'FirebaseDynamicLink#onLinkSuccess':
+          Map<String, dynamic> event =
+              Map<String, dynamic>.from(call.arguments);
+          _onLinkController.add(getPendingDynamicLinkDataFromMap(event));
+          break;
+        case 'FirebaseDynamicLink#onLinkError':
+          Map<String, dynamic> error =
+              Map<String, dynamic>.from(call.arguments);
+          _onLinkController.addError(convertPlatformException(error));
+          break;
+        default:
+          throw UnimplementedError('${call.method} has not been implemented');
+      }
+    });
+  }
+
+  /// The [FirebaseApp] instance to which this [FirebaseDynamicLinks] belongs.
+  ///
+  /// If null, the default [FirebaseApp] is used.
+
+  /// The [MethodChannel] used to communicate with the native plugin
+  static MethodChannel channel = const MethodChannel(
+    'plugins.flutter.io/firebase_dynamic_links',
+  );
+
+  /// The [StreamController] used to update on the latest dynamic link received.
+  final StreamController<PendingDynamicLinkData?> _onLinkController =
+      StreamController<PendingDynamicLinkData?>.broadcast();
+
+  /// Gets a [FirebaseDynamicLinksPlatform] with specific arguments such as a different
+  /// [FirebaseApp].
+  @override
+  FirebaseDynamicLinksPlatform delegateFor({required FirebaseApp app}) {
+    return MethodChannelFirebaseDynamicLinks(app: app);
+  }
+
+  /// Attaches generic default values to method channel arguments to allow multi-app support for android.
+  Map<String, dynamic> _withChannelDefaults(Map<String, dynamic> other) {
+    return {
+      'appName': appInstance?.name ?? defaultFirebaseAppName,
+    }..addAll(other);
+  }
+
+  PendingDynamicLinkData? getPendingDynamicLinkDataFromMap(
+    Map<dynamic, dynamic>? linkData,
+  ) {
+    if (linkData == null) return null;
+
+    final link = linkData['link'];
+    if (link == null) return null;
+
+    PendingDynamicLinkDataAndroid? androidData;
+    if (linkData['android'] != null) {
+      final Map<dynamic, dynamic> data = linkData['android'];
+      androidData = PendingDynamicLinkDataAndroid(
+        clickTimestamp: data['clickTimestamp'],
+        minimumVersion: data['minimumVersion'],
+      );
+    }
+
+    PendingDynamicLinkDataIOS? iosData;
+    if (linkData['ios'] != null) {
+      final Map<dynamic, dynamic> data = linkData['ios'];
+      iosData =
+          PendingDynamicLinkDataIOS(minimumVersion: data['minimumVersion']);
+    }
+
+    return PendingDynamicLinkData(
+      link: Uri.parse(link),
+      android: androidData,
+      ios: iosData,
+    );
+  }
+
+  @override
+  Future<PendingDynamicLinkData?> getInitialLink() async {
+    try {
+      final Map<String, dynamic>? linkData =
+          await channel.invokeMapMethod<String, dynamic>(
+        'FirebaseDynamicLinks#getInitialLink',
+        _withChannelDefaults({}),
+      );
+
+      return getPendingDynamicLinkDataFromMap(linkData);
+    } on PlatformException catch (e, s) {
```

Comment:
These should catch all errors (not just PlatFormExecption) - the `convertPlatformException` function handles this case.